exporters/webhook/plugins/pelorus_handler_base.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+#
+# Copyright Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+import http
+from abc import ABC, abstractmethod
+from json import JSONDecodeError
+from typing import Any, Awaitable, Optional
+
+from fastapi import HTTPException as FastapiHTTPException
+from pydantic import BaseModel
+from starlette.datastructures import Headers as StarletteHeaders
+from starlette.requests import Request as StarletteRequest
+
+from webhook.models.pelorus_webhook import PelorusMetric
+
+
+class HTTPException(FastapiHTTPException):
```

Comment:
did not understand these classes

Reply:
I think its unnecessary

Reply:
Having those classes in our base class of the plugin:
 - We simplify imports in the plugins. The plugin does not need to import the classes from other libraries, which can simplify the import statements and make the code more readable.
 - We have abstracted implementation: By creating those classes that inherit from starlette or fastapi library classes, the implementation details of those classes are abstracted away, making it easier to swap out the implementation or make changes without affecting the plugin code. So we can decide to use other frameworks and keep the plugins compatible.
 - By creating those classes within the plugin, the plugin can better encapsulate the functionality it provides.
 
From the cons I can think of just a bit more code, but we don't implement body of those classes so we don't have visible performance impact nor maintenance overhead.